04_setup_ironic.sh

```diff
@@ -28,36 +28,20 @@ done
 
 pushd $IRONIC_DATA_DIR/html/images
 
-# Workaround so that the dracut network module does dhcp on eth0 & eth1
-RHCOS_IMAGE_FILENAME_RAW="${RHCOS_IMAGE_FILENAME_OPENSTACK}.raw"
-if [ ! -e "$RHCOS_IMAGE_FILENAME_DUALDHCP" ] ; then
-    # Calculate the disksize required for the partitions on the image
-    # we do this to reduce the disk size so that ironic doesn't have to write as
-    # much data during deploy, as the default upstream disk image is way bigger
-    # then it needs to be. Were are adding the partition sizes and multiplying by 1.2.
-    DISKSIZE=$(virt-filesystems -a "$RHCOS_IMAGE_FILENAME_OPENSTACK" -l | grep /dev/ | awk '{s+=$5} END {print s*1.2}')
-    truncate --size $DISKSIZE "${RHCOS_IMAGE_FILENAME_RAW}"
-    virt-resize --no-extra-partition "${RHCOS_IMAGE_FILENAME_OPENSTACK}" "${RHCOS_IMAGE_FILENAME_RAW}"
-
-    LOOPBACK=$(sudo losetup --show -f "${RHCOS_IMAGE_FILENAME_RAW}" | cut -f 3 -d /)
-    mkdir -p /tmp/mnt
-    sudo kpartx -a /dev/$LOOPBACK
-    sudo mount /dev/mapper/${LOOPBACK}p1 /tmp/mnt
-    sudo sed --follow-symlinks -i -e 's/ip=eth0:dhcp/ip=eth0:dhcp ip=eth1:dhcp/g' /tmp/mnt/grub2/grub.cfg
-    sudo umount /tmp/mnt
-    sudo kpartx -d /dev/${LOOPBACK}
-    sudo losetup -d /dev/${LOOPBACK}
-    qemu-img convert -O qcow2 -c "$RHCOS_IMAGE_FILENAME_RAW" "$RHCOS_IMAGE_FILENAME_DUALDHCP"
-    rm "$RHCOS_IMAGE_FILENAME_RAW"
+RHCOS_IMAGE_FILENAME_COMPRESSED=$RHCOS_IMAGE_FILENAME_BM_BIOS.qcow2
+if [ ! -e "$RHCOS_IMAGE_FILENAME_COMPRESSED" ] ; then
+    qemu-img convert -O qcow2 -c "$RHCOS_IMAGE_FILENAME_BM_BIOS" "$RHCOS_IMAGE_FILENAME_COMPRESSED"
```

Comment:
Not sure why would `qemu-img convert` would be required here - isn't $RHCOS_IMAGE_FILENAME_BM_BIOS.qcow2 already decompressed?

Reply:
this is done to make the qcow2 file smaller, so it fits into /tmp on the masters (when Ironic is deploying it)